var/spack/repos/builtin/packages/hepmc3/package.py

```diff
@@ -85,4 +85,7 @@ def cmake_args(self):
                     self.define("HEPMC3_CXX_STANDARD", spec["root"].variants["cxxstd"].value)
                 )
 
+        if spec.satisfies("+protobuf"):
+            args.append(self.define("HEPMC3_CXX_STANDARD", "14"))
```

Comment:
Perhaps this should be an `elif`, otherwise it will collide with the ROOT definition? Perhaps also we should add a `cxxstd` variant that must match ROOT (if there) or be at least 14 if protobuf is included.

Reply:
yep, good point, agreed. I suspect any recent build including ROOT will set the standard to 17 at least, which obviates this. Its only for people (me) who want protobuf and not root support that it shows up. The real fix is on the HepMC3 cmake side, which I'll try and do when I have a spare cycle, but for now I think the elif covers most cases and still fixes my specific error.

Thanks!